service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DiaryPlanStatementMapper.java

```diff
@@ -0,0 +1,147 @@
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DiaryPlanStatementMapper {
+
+    private static final Mustache PLAN_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_plan_statement_template.mustache");
+    private static final String COMMA = ", ";
+    private static final String EMPTY_DATE = "nullFlavor=\"UNK\"";
+    private static final String FULL_DATE = "value=\"%s\"";
+
+    private final MessageContext messageContext;
+
+    public String mapDiaryProcedureRequestToPlanStatement(ProcedureRequest procedureRequest, Boolean isNested) {
+        if (procedureRequest.getIntent() != ProcedureRequest.ProcedureRequestIntent.PLAN) {
+            return StringUtils.EMPTY;
+        }
+
+        PlanStatementMapperParameters.PlanStatementMapperParametersBuilder builder = PlanStatementMapperParameters.builder()
+            .isNested(isNested)
+            .id(messageContext.getIdMapper().getOrNew(ResourceType.ProcedureRequest, procedureRequest.getId()))
+            .availabilityTime(buildAvailabilityTime(procedureRequest));
+
+        buildEffectiveTime(procedureRequest).map(builder::effectiveTime);
+        buildText(procedureRequest).map(builder::text);
+
+        return TemplateUtils.fillTemplate(PLAN_STATEMENT_TEMPLATE, builder.build());
+    }
+
+    private Optional<String> buildEffectiveTime(ProcedureRequest procedureRequest) {
+        Date date = null;
+        if (procedureRequest.hasOccurrenceDateTimeType()) {
+            date = procedureRequest.getOccurrenceDateTimeType().getValue();
+        } else if (procedureRequest.hasOccurrencePeriod()) {
+            Period occurrencePeriod = procedureRequest.getOccurrencePeriod();
+            date = occurrencePeriod.hasEnd() ? occurrencePeriod.getEnd() : occurrencePeriod.getStart();
+        }
+
+        return Optional.of(formatEffectiveDate(date));
+    }
+
+    private String formatEffectiveDate(Date date) {
+        return date != null ? String.format(FULL_DATE, DateFormatUtil.formatDate(date)) : EMPTY_DATE;
+    }
+
+    private String buildAvailabilityTime(ProcedureRequest procedureRequest) {
+        if (procedureRequest.hasAuthoredOn()) {
+            return DateFormatUtil.formatDate(procedureRequest.getAuthoredOn());
+        }
+
+        throw new EhrMapperException(
+            String.format("ProcedureRequest: %s does not have required authoredOn", procedureRequest.getId())
+        );
+    }
+
+    private Optional<String> buildText(ProcedureRequest procedureRequest) {
+        return Stream.of(
+            getRequester(procedureRequest),
+            getReasonCode(procedureRequest),
+            getNotes(procedureRequest)
+        )
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .reduce(joining());
+    }
+
+    private Optional<String> getReasonCode(ProcedureRequest procedureRequest) {
```

Comment:
Reason is not output at a concatenation of all text and display fields. It is just the first instance.

> "Reason Code: { reasonCode[0].text if populated then reasonCode[0].coding[0].display if populated }"

> 

Only supporting   the 1st reason. Not doing an open ended thing
--